src/components/GoogleMapsLink.js

```diff
@@ -12,19 +12,19 @@ class GoogleMapsLink extends React.Component {
                 <div>Link text</div>
             ),
             to: new Location(fixtures.ixa.location),
+            from: "",
         },
     };
 
     googleMapsUrl(): string {
         var toAddr = this.props.to;
 
-        // FIXME: this.props.from
-        var start = "Current+Location";
+        var start = encodeURIComponent(this.props.from || "");
         var query = encodeURIComponent(
             `${toAddr.streetAddressLine1()} ${toAddr.streetAddressLine2()}`
         );
 
-        return `https://maps.google.com/?saddr=${start}&addr=${query}`;
+        return `https://maps.google.com/?directionsmode=walking&saddr=${start}&daddr=${query}`;
```

Comment:
Let's add a FIXME to do it based on whichever we're showing in the travel time widget once that works.